webapp/components/backstage/installed_commands.jsx

```diff
@@ -21,16 +22,18 @@ export default class InstalledCommands extends React.Component {
         this.deleteCommand = this.deleteCommand.bind(this);
 
         this.state = {
-            commands: IntegrationStore.getCommands(),
-            loading: !IntegrationStore.hasReceivedCommands()
+            commands: IntegrationStore.getCommands(TeamStore.getCurrentId()),
+            loading: true
         };
     }
 
     componentDidMount() {
         IntegrationStore.addChangeListener(this.handleIntegrationChange);
 
-        if (window.mm_config.EnableCommands === 'true' && this.state.loading) {
-            AsyncClient.listTeamCommands();
+        if (window.mm_config.EnableCommands === 'true') {
+            AsyncClient.listTeamCommands(() => {
```

Comment:
Yeah, I wasn't thinking right about this. I'll re-add something similar to the previous IntegrationStore.hasCommands and set this.state.loading without a callback